sync/src/main/kotlin/gropius/sync/AbstractSync.kt

```diff
@@ -282,16 +290,29 @@ abstract class AbstractSync(
             issue = dereplicationResult.resultingIssue
             for (fakeSyncedItem in dereplicationResult.fakeSyncedItems) {
                 nodesToSave.add(fakeSyncedItem)
-                savedNodeHandlers.add {
-                    val tici = DummyTimelineItemConversionInformation(imsProject.rawId!!, (it as TimelineItem).rawId!!)
-                    tici.gropiusId = (it as TimelineItem).rawId
+                savedNodeHandlers.add { updatedNode ->
+                    val tici = DummyTimelineItemConversionInformation(
+                        imsProject.rawId!!, (updatedNode as TimelineItem).rawId!!
+                    )
+                    tici.gropiusId = (updatedNode as TimelineItem).rawId
                     collectedSyncInfo.timelineItemConversionInformationService.save(tici).awaitSingle()
                 }
             }
         }
         val savedList = collectedSyncInfo.neoOperations.saveAll(nodesToSave).collectList().awaitSingle()
         val savedIssue = savedList.removeFirst()
         if (issue.rawId == null) issue = savedIssue as Issue
+        val updater = IssueAggregationUpdater()
+        if (isNewIssue) {
+            updater.addedIssueToTrackable(issue, imsProject.trackable().value)
+        }
+        updater.changedIssueStateOrType(
+            issue,
+            (if (oldState.rawId != null) collectedSyncInfo.neoOperations.findById<IssueState>(oldState.rawId!!) else null)
```

Comment:
Why not directly else oldState?
Also, is this reloading required or did you just add it for safety?

Reply:
Because if a StateChangedTimelineItem was inserted, it will have touched the oldState and thus made the old state invalid

Reply:
0495a0e6fe8a26054ac4908f7a1ce8e0011c8a87